host/windows/syscall.c

```diff
@@ -20,16 +19,163 @@
+    {WSAEWOULDBLOCK, OE_EWOULDBLOCK},
+    {WSAEINPROGRESS, OE_EINPROGRESS},
+    {WSAEALREADY, OE_EALREADY},
+    {WSAENOTSOCK, OE_ENOTSOCK},
+    {WSAEDESTADDRREQ, OE_EDESTADDRREQ},
+    {WSAEMSGSIZE, OE_EMSGSIZE},
+    {WSAEPROTOTYPE, OE_EPROTOTYPE},
+    {WSAENOPROTOOPT, OE_ENOPROTOOPT},
+    {WSAEPROTONOSUPPORT, OE_EPROTONOSUPPORT},
+    {WSAESOCKTNOSUPPORT, OE_ESOCKTNOSUPPORT},
+    {WSAEOPNOTSUPP, OE_EOPNOTSUPP},
+    {WSAEPFNOSUPPORT, OE_EPFNOSUPPORT},
+    {WSAEAFNOSUPPORT, OE_EAFNOSUPPORT},
+    {WSAEADDRINUSE, OE_EADDRINUSE},
+    {WSAEADDRNOTAVAIL, OE_EADDRNOTAVAIL},
+    {WSAENETDOWN, OE_ENETDOWN},
+    {WSAENETUNREACH, OE_ENETUNREACH},
+    {WSAENETRESET, OE_ENETRESET},
+    {WSAECONNABORTED, OE_ECONNABORTED},
+    {WSAECONNRESET, OE_ECONNRESET},
+    {WSAENOBUFS, OE_ENOBUFS},
+    {WSAEISCONN, OE_EISCONN},
+    {WSAENOTCONN, OE_ENOTCONN},
+    {WSAESHUTDOWN, OE_ESHUTDOWN},
+    {WSAETOOMANYREFS, OE_ETOOMANYREFS},
+    {WSAETIMEDOUT, OE_ETIMEDOUT},
+    {WSAECONNREFUSED, OE_ECONNREFUSED},
+    {WSAELOOP, OE_ELOOP},
+    {WSAENAMETOOLONG, OE_ENAMETOOLONG},
+    {WSAEHOSTDOWN, OE_EHOSTDOWN},
+    {WSAEHOSTUNREACH, OE_EHOSTUNREACH},
+    {WSAENOTEMPTY, OE_ENOTEMPTY},
+    {WSAEUSERS, OE_EUSERS},
+    {WSAEDQUOT, OE_EDQUOT},
+    {WSAESTALE, OE_ESTALE},
+    {WSAEREMOTE, OE_EREMOTE},
+    {WSAEDISCON, 199},
+    {WSAEPROCLIM, 200},
+    {WSASYSNOTREADY, 201}, // Made up number but close to adjacent
+    {WSAVERNOTSUPPORTED, 202},
+    {WSANOTINITIALISED, 203},
+    {0, 0}};
+
+/**
+ * Musl libc has redefined pretty much every define in socket.h so that
+ * constants passed as parameters are different if the enclave uses musl
+ * and the host uses a socket implementation that uses the original BSD
+ * defines (winsock, glibc, BSD libc). The following tables are 1-to-1 mappings
+ * from musl defines to bsd defines
+ */
+
+// Only SOL_SOCKET is different. All other socket level
+// defines are the same.
+static struct tab_entry musl2bsd_socket_level[] = {{1, SOL_SOCKET}, {0, 0}};
+
+static struct tab_entry musl2bsd_socket_option[] = {{1, SO_DEBUG},
+                                                    {2, SO_REUSEADDR},
+                                                    {3, SO_TYPE},
+                                                    {4, SO_ERROR},
+                                                    {5, SO_DONTROUTE},
+                                                    {6, SO_BROADCAST},
+                                                    {7, SO_SNDBUF},
+                                                    {8, SO_RCVBUF},
+                                                    {9, SO_KEEPALIVE},
+                                                    {10, SO_OOBINLINE},
+                                                    {13, SO_LINGER},
+                                                    {18, SO_RCVLOWAT},
+                                                    {19, SO_SNDLOWAT}};
+
+// From netdb.h
+#define EAI_BADFLAGS -1
```

Comment:
These defines are already defined in ws2tcpip.h with different values, so this is confusing in a Windows file.
I don't think we should redefine standard values here in this file.  If you need to have different numerical values for EAI_MEMORY etc than Windows defines, then I think this should have a different name, like OE_EAI_MEMORY or something.